monai/transforms/utils.py

```diff
@@ -1970,5 +1975,83 @@ def resolves_modes(
     return backend, _interp_mode, _padding_mode, _kwargs
 
 
+def check_applied_operations(entry: list | dict, status_key: str, default_message: str = "No message provided"):
+    """
+    Check the operations of a MetaTensor to determine whether there are any statuses
+    Args:
+        entry: a dictionary that may contain TraceKey.STATUS entries, or a list of such dictionaries
+        status_key: the status key to search for. This must be an entry in `TraceStatusKeys`_
+        default_message: The message to provide if no messages are provided for the given status key entry
+
+    Returns:
+        A list of status messages matching the providing status key
+
+    """
+    if isinstance(entry, list):
+        results = list()
+        for sub_entry in entry:
+            results.extend(check_applied_operations(sub_entry, status_key, default_message))
+        return results
+    else:
+        status_key_ = TraceStatusKeys(status_key)
+        if TraceKeys.STATUSES in entry:
+            if status_key_ in entry[TraceKeys.STATUSES]:
+                reason = entry[TraceKeys.STATUSES][status_key_]
+                if reason is None:
+                    return [default_message]
+                return reason if isinstance(reason, list) else [reason]
+        return []
+
+
+def is_tensor_invertible(data: torch.Tensor):
```

Comment:
I think this function introduced a new concept: "data is invertible", we only discuss "transform is invertible" in the MONAI codebase so far.
I think we don't need to check whether a data is invertible (and it looks confusing), we just need to check: "whether it has pending operations".
And let's avoid adding new concept in the last-minute change.
What do you think? CC @ericspod .

Thanks.

Reply:
That's exactly what happened on dev. We got the new mechanism with the out of order execution once the problem was detected. All I did was move it from info to a new flag.

Reply:
I think this one is similar to what @ericspod described in the `check_data` comment, better to define function / var names reflecting the exact meaning of this component in the end-to-end workflow.
And I think we don't need to consider `dev` and `out of order execution`, `dev` is not released and supposed to be optimized by this PR, we just need to do the right and simple things.
`has_pending_operations(data: torch.Tensor):` looks more clear to me.
@ericspod What do you think?

Thanks.

Reply:
I checked the doc-string:
```
1. If the tensor is not a MetaTensor, it is not invertible
2. If the tensor is a MetaTensor but it has `TraceStatusKeys.PENDING_DURING_APPLY` in the `TraceKeys.STATUS` of any of its `applied_operations` it is not invertible
3. Otherwise, it is invertible
```
Seems only "has pending operations" is "not invertible" for a `MetaTensor`, it's the deep reason of the function name "is tensor invertible".
So I think we don't need to explicitly introduce this new concept "data is invertible or not", just need to check whether a `MetaTensor` still has pending operations before inverting its executed transforms.

Thanks.

Reply:
That wouldn't be enough. Having `pending_transforms` at the end, ironically, wouldn't break the `inverse` code as it is implemented at the moment. The issue arises in the case where, _at some point in the pipeline_, there _were_ pending transforms at the point that a transform was applied. That information is lost by the end of the pipeline unless you track it somehow.

Reply:
Renamed  to `has_status_keys`